src/databricks/labs/ucx/source_code/base.py

```diff
@@ -214,7 +214,7 @@ def replace_source(
         source_id: str | None = None,
         source_lineage: list[LineageAtom] | None = None,
         source_timestamp: datetime | None = None,
-    ):
+    ) -> Self:
```

Comment:
`Self` isn't available from `typing` in python 3.10, and `typing_extensions` (where it comes from here) is a transitive dependency. If we're using this directly in our code I'd prefer `typing_extensions` to be on our list of project dependencies rather than relying on it transitively.